tests/cephadm/test_custom_container.py

```diff
@@ -0,0 +1,272 @@
+from json import load
+
+from ceph.waiter import WaitUntil
+from cli.cephadm.cephadm import CephAdm
+from utility.utils import Log, get_cephci_config
+
+log = Log(__name__)
+
+PRIVATE_REG_USER = "myregistryusername"
+PRIVATE_REG_PASS = "myregistrypassword1"
+OPT_AUTH = "/opt/registry/auth"
+OPT_CERTS = "/opt/registry/certs"
+OPT_DATA = "/opt/registry/data"
+
+
+class ClusterCustomContainerError(Exception):
+    pass
+
+
+def run(ceph_cluster, **kw):
+    """Verify admin node has updated view of cluster
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get installer node
+    node = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Get config file and node
+    config = kw.get("config")
+    ceph_nodes = kw.get("ceph_nodes")
+
+    # Install httpsd-tools and httpd
+    node.exec_command(
+        sudo=True,
+        cmd="dnf install -y httpd-tools httpd",
+        long_running=True,
+    )
+
+    # Create folders for registry
+    node.exec_command(sudo=True, cmd="mkdir -p /opt/registry/{auth,certs,data}")
+
+    # Create credentials for accessing the private registry
+    node.exec_command(
+        sudo=True,
+        cmd=f"htpasswd -bBc {OPT_AUTH}/htpasswd \
+             {PRIVATE_REG_USER} {PRIVATE_REG_PASS}",
+    )
+
+    # Create a self-signed certificate
+    node.exec_command(
```

Comment:
are echo cmd required here? Just '.' won't work? Please ignore if you have already tested

Reply:
It works as expected.